### PR TITLE
Update getAliasThis documentation per issue 19439

### PR DESCRIPTION
**Pull request by n8sh**
Companion PR to https://github.com/dlang/dmd/pull/9016

### CODE REVIEW

File: spec/traits.dd [hunk 1]
```diff
@@ -630,12 +630,35 @@ writeln(__traits(identifier, var));             // "var"
 
 $(H2 $(GNAME getAliasThis))
 
-    $(P Takes one argument, a symbol of aggregate type.
-        If the given aggregate type has $(D alias this), returns a list of
-        $(D alias this) names, by a tuple of $(D string)s.
-        Otherwise returns an empty tuple.
+    $(P Takes one argument, a symbol. If the symbol is of a type which has
+        $(D alias this), returns a list of $(D alias this) names, by a tuple
```
Comment (mdparker) [inline spec/traits.dd]: names, by a tuple -> names as a tuple

Comment (ZombineDev) [reply spec/traits.dd]: The name we use to call what `AliasSeq` represents was debated at length in the newsgroup a couple years ago and we agreed that we should call them "sequences" to distinguish them from tuple-like objects like the one defined in `std.typecons.Tuple`.

The difference is the following:
* Tuples are value-types holding other run-time specified values, whose types are fixed at compile-time, i.e. at run-time you can change value of the i-th element, but not its type. Nesting is respected, i.e. `Tuple!(int, Tuple!(Tuple!double)) != Tuple!(int, double)`.

* Sequences are lists of compile-time entities (which can include literal values, types, and symbols). They are immutable and they don't nest, i.e. `AliasSeq!(int, AliasSeq!(AliasSeq!double)) == AliasSeq!(int, double)`

So, in summary I recommend the following wording:

```suggestion
        Takes one argument, a type. If the type has `alias this` declaration(s),
        returns the sequence members (represented as `string`s) which were
        part of those declarationd. Otherwise returns an empty sequence.
```

File: spec/traits.dd [hunk 2]
```diff
@@ -630,12 +630,35 @@ writeln(__traits(identifier, var));             // "var"
...
+        of $(D string)s. Otherwise returns an empty tuple.
     )
 
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+---
+alias TypeTuple(T...) = T;
```
Comment (ZombineDev) [inline spec/traits.dd]: As mentioned above, `TypeTuple` is the wrong name, as it's neither a tuple, nor in the example below holds any types.

```suggestion
alias AliasSeq(Args...) = Args;
```

Comment (n8sh) [reply spec/traits.dd]: Implemented.

File: spec/traits.dd (hunk 1 above)
Comment (n8sh) [reply spec/traits.dd]: I have not adopted the suggested wording exactly, because to me "returns the sequence members which [...]" sounds like we are selecting some members of an existing sequence. Please review whether the new wording is acceptable.
